cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -5039,6 +5057,49 @@ if(ARROW_S3)
   endif()
 endif()
 
+# ----------------------------------------------------------------------
+# Azure SDK for C++
+
+function(build_azure_sdk)
+  message(STATUS "Building Azure SDK for C++ from source")
+  fetchcontent_declare(azure_sdk
+                       URL ${ARROW_AZURE_SDK_URL}
+                       URL_HASH "SHA256=${ARROW_AZURE_SDK_BUILD_SHA256_CHECKSUM}")
+  fetchcontent_getproperties(azure_sdk)
+  if(NOT azure_sdk_POPULATED)
+    fetchcontent_populate(azure_sdk)
+    set(BUILD_PERFORMANCE_TESTS FALSE)
+    set(BUILD_SAMPLES FALSE)
+    set(BUILD_TESTING FALSE)
+    set(BUILD_WINDOWS_UWP TRUE)
+    set(CMAKE_EXPORT_NO_PACKAGE_REGISTRY TRUE)
+    set(DISABLE_AZURE_CORE_OPENTELEMETRY TRUE)
```

Disable vcpkg integration.

```suggestion
    set(DISABLE_AZURE_CORE_OPENTELEMETRY TRUE)
    set(ENV{AZURE_SDK_DISABLE_AUTO_VCPKG} TRUE)
```